blockchain/v1/reactor_fsm.go

```diff
@@ -269,7 +275,13 @@ func init() {
 					return waitForPeer, err
 				}
 				return waitForBlock, err
+			case noBlockResponseEv:
+				fsm.logger.Error("peer does not have requested block", "peer", data.peerID, "height", data.height)
+				fsm.pool.RemovePeer(data.peerID, fmt.Errorf("peer does not have block %d", data.height))
```

Comment:
This may be a bit harsh. If a peer is running with block pruning enabled, then they will continually be deleting their oldest block, so there is a race where they report a certain base, then prune it, and then we request the pruned height.

Reply:
This was my thinking as well. It is weird a tradeoff. V0 just logs an error while v2 removes the peer. It was unclear which approach should be taken. I am happy to only log the error instead of removing the peer.

Reply:
Do we know that v2 approach works in practice? v0 is battle tested.

Reply:
but neither is really battle tested with pruning right?

Reply:
Yeah, I'd say we should err on the side of caution and keep the peer. If it turns out to be a problem, we can change it later.

Reply:
> Do we know that v2 approach works in practice? v0 is battle tested.

No, it is unclear. Will go with logging of error for now

> but neither is really battle tested with pruning right?

Correct